### ISSUE TITLE
[GLPI 10] Error inserting email to queue 

### ISSUE THREAD

**Issue by Badkempachi** [glpi-project/glpi#10497]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10

### Bug description

E-mail alert activated when certificate expires.
The certificate was added yesterday afternoon, and after the "Automatic action - certificate" task was sent, and the error "Error inserting e-mail into queue" was received
Please help in solving the problem.

Also, if possible, will be delayed about adding a new certificate. The addendum has a "Validity" field in the description says "In the search engine, use "Validity contains NULL" to search for licenses without an expiration date" , but I did not find detailed information about this in the documentation. Interested in what cases require the use of this item?

### Relevant log output

```shell
2022-02-01 12:06:34] glpiphplog.WARNING: Test logger  
[2022-02-01 12:23:28] glpiphplog.WARNING: Test logger  
[2022-02-01 12:25:08] glpiphplog.NOTICE:   *** PHP Notice (8): Undefined index: users_id_tech in /var/www/html/glpi/src/NotificationTarget.php at line 1041
  Backtrace :
  src/NotificationTarget.php:1067                    NotificationTarget->addUserByField()
  src/NotificationTarget.php:1187                    NotificationTarget->addItemTechnicianInCharge()
  src/NotificationEventAbstract.php:93               NotificationTarget->addForTarget()
  src/NotificationEvent.php:185                      NotificationEventAbstract::raise()
  src/Certificate.php:750                            NotificationEvent::raiseEvent()
  src/CronTask.php:1016                              Certificate::cronCertificate()
  front/crontask.form.php:50                         CronTask::launch()
  
[2022-02-01 12:25:08] glpiphplog.NOTICE:   *** PHP Notice (8): Undefined index: groups_id_tech in /var/www/html/glpi/src/NotificationTarget.php at line 1080
  Backtrace :
  src/NotificationTarget.php:1192                    NotificationTarget->addItemGroupTechInCharge()
  src/NotificationEventAbstract.php:93               NotificationTarget->addForTarget()
  src/NotificationEvent.php:185                      NotificationEventAbstract::raise()
  src/Certificate.php:750                            NotificationEvent::raiseEvent()
  src/CronTask.php:1016                              Certificate::cronCertificate()
  front/crontask.form.php:50                         CronTask::launch()
  
[2022-02-01 12:30:31] glpiphplog.DEBUG: Html::compileScss() in /var/www/html/glpi/src/Html.php line 6697
Compile css/palettes/auror.scss  {"user":"9@glpi.localhost.ru"} 
[2022-02-01 12:32:46] glpiphplog.NOTICE:   *** PHP Notice (8): Undefined index: users_id_tech in /var/www/html/glpi/src/NotificationTarget.php at line 1041
  Backtrace :
  src/NotificationTarget.php:1067                    NotificationTarget->addUserByField()
  src/NotificationTarget.php:1187                    NotificationTarget->addItemTechnicianInCharge()
  src/NotificationEventAbstract.php:93               NotificationTarget->addForTarget()
  src/NotificationEvent.php:185                      NotificationEventAbstract::raise()
  src/Certificate.php:750                            NotificationEvent::raiseEvent()
  src/CronTask.php:1016                              Certificate::cronCertificate()
  front/crontask.form.php:50                         CronTask::launch()
  
[2022-02-01 12:32:46] glpiphplog.NOTICE:   *** PHP Notice (8): Undefined index: groups_id_tech in /var/www/html/glpi/src/NotificationTarget.php at line 1080
  Backtrace :
  src/NotificationTarget.php:1192                    NotificationTarget->addItemGroupTechInCharge()
  src/NotificationEventAbstract.php:93               NotificationTarget->addForTarget()
  src/NotificationEvent.php:185                      NotificationEventAbstract::raise()
  src/Certificate.php:750                            NotificationEvent::raiseEvent()
  src/CronTask.php:1016                              Certificate::cronCertificate()
  front/crontask.form.php:50                         CronTask::launch()
  
[2022-02-01 12:32:46] glpiphplog.NOTICE:   *** PHP Notice (8): Undefined index: usertype in /var/www/html/glpi/src/NotificationEventMailing.php at line 89
  Backtrace :
  src/NotificationTarget.php:655                     NotificationEventMailing::getAdminData()
  src/NotificationTarget.php:1174                    NotificationTarget->addAdmin()
  src/NotificationEventAbstract.php:93               NotificationTarget->addForTarget()
  src/NotificationEvent.php:185                      NotificationEventAbstract::raise()
  src/Certificate.php:750                            NotificationEvent::raiseEvent()
  src/CronTask.php:1016                              Certificate::cronCertificate()
  front/crontask.form.php:50                         CronTask::launch()
  
[2022-02-01 12:35:35] glpiphplog.NOTICE:   *** PHP Notice (8): Undefined index: users_id_tech in /var/www/html/glpi/src/NotificationTarget.php at line 1041
  Backtrace :
  src/NotificationTarget.php:1067                    NotificationTarget->addUserByField()
  src/NotificationTarget.php:1187                    NotificationTarget->addItemTechnicianInCharge()
  src/NotificationEventAbstract.php:93               NotificationTarget->addForTarget()
  src/NotificationEvent.php:185                      NotificationEventAbstract::raise()
  src/Certificate.php:750                            NotificationEvent::raiseEvent()
  src/CronTask.php:1016                              Certificate::cronCertificate()
  front/crontask.form.php:50                         CronTask::launch()
  
[2022-02-01 12:35:35] glpiphplog.NOTICE:   *** PHP Notice (8): Undefined index: groups_id_tech in /var/www/html/glpi/src/NotificationTarget.php at line 1080
  Backtrace :
  src/NotificationTarget.php:1192                    NotificationTarget->addItemGroupTechInCharge()
  src/NotificationEventAbstract.php:93               NotificationTarget->addForTarget()
  src/NotificationEvent.php:185                      NotificationEventAbstract::raise()
  src/Certificate.php:750                            NotificationEvent::raiseEvent()
  src/CronTask.php:1016                              Certificate::cronCertificate()
  front/crontask.form.php:50                         CronTask::launch()
  
[2022-02-01 12:35:35] glpiphplog.NOTICE:   *** PHP Notice (8): Undefined index: usertype in /var/www/html/glpi/src/NotificationEventMailing.php at line 89
  Backtrace :
  src/NotificationTarget.php:655                     NotificationEventMailing::getAdminData()
  src/NotificationTarget.php:1174                    NotificationTarget->addAdmin()
  src/NotificationEventAbstract.php:93               NotificationTarget->addForTarget()
  src/NotificationEvent.php:185                      NotificationEventAbstract::raise()
  src/Certificate.php:750                            NotificationEvent::raiseEvent()
  src/CronTask.php:1016                              Certificate::cronCertificate()
  front/crontask.form.php:50                         CronTask::launch()
```

### Steps To reproduce

![1](https://user-images.githubusercontent.com/97535165/151950080-8ddef316-1283-4ffd-8b58-8be9be7bc49d.png)
![2](https://user-images.githubusercontent.com/97535165/151950085-6af9459e-bdc9-4558-92a0-cfbd483c0a26.png)
![3](https://user-images.githubusercontent.com/97535165/151950087-b47fa50a-2470-4375-90b5-e8f38c41a91e.png)
![4](https://user-images.githubusercontent.com/97535165/151950089-a7574fcf-d528-466d-ab3b-75dd90affe0c.png)
![5](https://user-images.githubusercontent.com/97535165/151950093-d4400914-b90f-47a0-9706-fd714abdca41.png)
![6](https://user-images.githubusercontent.com/97535165/151950096-7ae4fb89-edcd-4d49-984c-c7c0a1015216.png)

### Your GLPI setup information

<details><summary>Information about system installation and configuration</summary><pre>GLPI 10.0.0-dev ( => /var/www/html/glpi)
Installation mode: TARBALL
Current language:en_GB

</details><details><summary>Server</summary><pre> 
Operating system: Linux glpi.localhost.ru 5.4.17-2136.302.7.2.1.el8uek.x86_64 #2 SMP Tue Jan 18 12:11:34 PST 2022 x86_64
PHP 7.4.27 fpm-fcgi (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apcu, bz2, calendar, cgi-fcgi, ctype, curl,
	date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, intl, json, ldap, libxml, mbstring, mysqli, mysqlnd, openssl,
	pcre, pdo_mysql, pdo_sqlite, session, sockets, sodium, sqlite3, standard, tokenizer, xml, xmlreader, xmlrpc, xmlwriter, xsl,
	zip, zlib)
Setup: max_execution_time="30" memory_limit="256M" post_max_size="8M" safe_mode="" session.save_handler="files"
	upload_max_filesize="2M" 
Software: Apache/2.4.37 (Oracle Linux) OpenSSL/1.1.1k ()
	Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:91.0) Gecko/20100101 Firefox/91.0
Server Software: MariaDB Server
	Server Version: 10.3.28-MariaDB
	Server SQL Mode: STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
	Parameters: root@localhost/GLPI
	Host info: Localhost via UNIX socket
	
PHP version (7.4.27) is supported.
Sessions configuration is OK.
Allocated memory is sufficient.
mysqli extension is installed.
Following extensions are installed: dom, fileinfo, json, simplexml.
curl extension is installed.
gd extension is installed.
intl extension is installed.
libxml extension is installed.
zlib extension is installed.
Database engine version (10.3.28) is supported.
The log file has been created successfully.
Write access to /var/www/html/glpi/files/_cache has been validated.
Write access to /var/www/html/glpi/config has been validated.
Write access to /var/www/html/glpi/files/_cron has been validated.
Write access to /var/www/html/glpi/files has been validated.
Write access to /var/www/html/glpi/files/_dumps has been validated.
Write access to /var/www/html/glpi/files/_graphs has been validated.
Write access to /var/www/html/glpi/files/_lock has been validated.
Write access to /var/www/html/glpi/files/_pictures has been validated.
Write access to /var/www/html/glpi/files/_plugins has been validated.
Write access to /var/www/html/glpi/files/_rss has been validated.
Write access to /var/www/html/glpi/files/_sessions has been validated.
Write access to /var/www/html/glpi/files/_tmp has been validated.
Write access to /var/www/html/glpi/files/_uploads has been validated.
Web access to the files directory should not be allowed but this cannot be checked automatically on this instance.
Make sure access to error log file (/files/_log/php-errors.log) is forbidden; otherwise review .htaccess file and web server configuration.
SELinux configuration is OK.
exif extension is installed.
ldap extension is installed.
openssl extension is installed.
zip extension is installed.
bz2 extension is installed.
Zend OPcache extension is installed.
Following extensions are installed: ctype, iconv, mbstring, sodium.
Write access to /var/www/html/glpi/marketplace has been validated.
Timezones seems loaded in database.

</details><details><summary>GLPI constants</summary><pre> 
GLPI_ROOT: /var/www/html/glpi
GLPI_CONFIG_DIR: /var/www/html/glpi/config
GLPI_VAR_DIR: /var/www/html/glpi/files
GLPI_MARKETPLACE_DIR: /var/www/html/glpi/marketplace
GLPI_USE_CSRF_CHECK: 1
GLPI_CSRF_EXPIRES: 7200
GLPI_CSRF_MAX_TOKENS: 100
GLPI_USE_IDOR_CHECK: 1
GLPI_IDOR_EXPIRES: 7200
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: 
GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org
GLPI_INSTALL_MODE: TARBALL
GLPI_NETWORK_MAIL: glpi@teclib.com
GLPI_NETWORK_SERVICES: https://services.glpi-network.com
GLPI_MARKETPLACE_ALLOW_OVERRIDE: 1
GLPI_MARKETPLACE_MANUAL_DOWNLOADS: 1
GLPI_USER_AGENT_EXTRA_COMMENTS: 
GLPI_DISABLE_ONLY_FULL_GROUP_BY_SQL_MODE: 1
GLPI_AJAX_DASHBOARD: 1
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: 0
GLPI_CENTRAL_WARNINGS: 1
GLPI_DOC_DIR: /var/www/html/glpi/files
GLPI_CACHE_DIR: /var/www/html/glpi/files/_cache
GLPI_CRON_DIR: /var/www/html/glpi/files/_cron
GLPI_DUMP_DIR: /var/www/html/glpi/files/_dumps
GLPI_GRAPH_DIR: /var/www/html/glpi/files/_graphs
GLPI_LOCAL_I18N_DIR: /var/www/html/glpi/files/_locales
GLPI_LOCK_DIR: /var/www/html/glpi/files/_lock
GLPI_LOG_DIR: /var/www/html/glpi/files/_log
GLPI_PICTURE_DIR: /var/www/html/glpi/files/_pictures
GLPI_PLUGIN_DOC_DIR: /var/www/html/glpi/files/_plugins
GLPI_RSS_DIR: /var/www/html/glpi/files/_rss
GLPI_SESSION_DIR: /var/www/html/glpi/files/_sessions
GLPI_TMP_DIR: /var/www/html/glpi/files/_tmp
GLPI_UPLOAD_DIR: /var/www/html/glpi/files/_uploads
GLPI_INVENTORY_DIR: /var/www/html/glpi/files/_inventories
GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/
GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/
GLPI_I18N_DIR: /var/www/html/glpi/locales
GLPI_VERSION: 10.0.0-dev
GLPI_SCHEMA_VERSION: 10.0.0-dev@1c2d91725deb0608a95fc36f9e27f9988fb5a89a
GLPI_MARKETPLACE_PRERELEASES: 1
GLPI_MIN_PHP: 7.4.0
GLPI_MAX_PHP: 8.2.0
GLPI_YEAR: 2021

</details><details><summary>Libraries</summary><pre> 
htmlawed/htmlawed version 1.2.6 in (/var/www/html/glpi/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.5.3 in (/var/www/html/glpi/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.8 in (/var/www/html/glpi/vendor/simplepie/simplepie/library)
mpdf/mpdf in (/var/www/html/glpi/vendor/mpdf/mpdf/src)
michelf/php-markdown in (/var/www/html/glpi/vendor/michelf/php-markdown/Michelf)
true/punycode in (/var/www/html/glpi/vendor/true/punycode/src)
iamcal/lib_autolink in (/var/www/html/glpi/vendor/iamcal/lib_autolink)
sabre/dav in (/var/www/html/glpi/vendor/sabre/dav/lib/DAV)
sabre/http in (/var/www/html/glpi/vendor/sabre/http/lib)
sabre/uri in (/var/www/html/glpi/vendor/sabre/uri/lib)
sabre/vobject in (/var/www/html/glpi/vendor/sabre/vobject/lib)
laminas/laminas-i18n in (/var/www/html/glpi/vendor/laminas/laminas-i18n/src)
laminas/laminas-servicemanager in (/var/www/html/glpi/vendor/laminas/laminas-servicemanager/src)
monolog/monolog in (/var/www/html/glpi/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/var/www/html/glpi/vendor/sebastian/diff/src)
donatj/phpuseragentparser in (/var/www/html/glpi/vendor/donatj/phpuseragentparser/src/UserAgent)
elvanto/litemoji in (/var/www/html/glpi/vendor/elvanto/litemoji/src)
symfony/console in (/var/www/html/glpi/vendor/symfony/console)
scssphp/scssphp in (/var/www/html/glpi/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/var/www/html/glpi/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/var/www/html/glpi/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/var/www/html/glpi/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/var/www/html/glpi/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/var/www/html/glpi/vendor/ramsey/uuid/src)
psr/log in (/var/www/html/glpi/vendor/psr/log/Psr/Log)
psr/simple-cache in (/var/www/html/glpi/vendor/psr/simple-cache/src)
psr/cache in (/var/www/html/glpi/vendor/psr/cache/src)
league/csv in (/var/www/html/glpi/vendor/league/csv/src)
mexitek/phpcolors in (/var/www/html/glpi/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/var/www/html/glpi/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/var/www/html/glpi/vendor/guzzlehttp/psr7/src)
glpi-project/inventory_format in (/var/www/html/glpi/vendor/glpi-project/inventory_format/lib/php)
wapmorgan/unified-archive in (/var/www/html/glpi/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/var/www/html/glpi/vendor/paragonie/sodium_compat/src)
symfony/cache in (/var/www/html/glpi/vendor/symfony/cache)
html2text/html2text in (/var/www/html/glpi/vendor/html2text/html2text/src)
symfony/dom-crawler in (/var/www/html/glpi/vendor/symfony/dom-crawler)
twig/twig in (/var/www/html/glpi/vendor/twig/twig/src)
twig/string-extra in (/var/www/html/glpi/vendor/twig/string-extra)
symfony/polyfill-ctype not found
symfony/polyfill-iconv not found
symfony/polyfill-mbstring not found
symfony/polyfill-php80 in (/var/www/html/glpi/vendor/symfony/polyfill-php80)

</details><details><summary>LDAP directories</summary><pre> 
Server: '192.168.1.1', Port: '389', BaseDN: 'DC=localhost,DC=ru', Connection filter:
		'(&(objectCategory=person)(objectClass=user)(!(userAccountControl:1.2.840.113556.1.4.803:=2)))', RootDN:
		'ldapsearch-glpi@localhost.ru', Use TLS: none

</details><details><summary>SQL replicas</summary><pre> 
Not active

</details><details><summary>Notifications</summary><pre> 
Way of sending emails: SMTP (anonymous@mail2.localhost.ru)

</details><details><summary>Plugins list</summary><pre> 
	connections          Name: Connections                    Version: 10.0.0-rc2 State: Enabled
	manageentities       Name: Entities portal                Version: 4.0.0-rc1  State: Enabled
	glpiinventory        Name: GLPI Inventory                 Version: 1.0.0-rc3  State: Enabled
	addressing           Name: IP Adressing                   Version: 3.0.0-rc2  State: Enabled
	mreporting           Name: Подробные отчеты          Version: 1.8.0-rc1  State: Enabled
	shellcommands        Name: Служебные команды        Version: 4.0.0-rc2  State: Enabled
	typology             Name: Типологий                      Version: 3.0.0-rc1  State: Enabled

</pre></details>

**Comment by Badkempachi** [comment]
![7](https://user-images.githubusercontent.com/97535165/151957519-4d2161a3-4b32-4795-9864-c93a6ebe5a3b.png)

**Comment by cconard96** [comment]
Related to #10507

The issue seems to be when trying to add recipients for group and technician in charge since this notification is sent for multiple certificates at once and not single items. The same issue may be present for other grouped notifications like consumable/cartridge alerts.

The workaround would be to change the notification recipients to something else.

**Comment by cedric-anne** [comment]
IMHO, having grouped notification and adding technician/group in charge in recipient are two features that should not be used together. Indeed, each item may be assigned to different tecnicians/groups, so a technician would sometimes receive a notification containing many elements that are not assigned to him.

#6799 was finally a bad idea, as it never worked.

There are many solutions to this.

1) Remove these targets (this is easy to do, but not having these targets may be considered as a missing feature).

2) Add all concerned technicians to the notification. It would require to change recipient types from `Notification::ITEM_TECH_IN_CHARGE` to a specific value, so we could handle this in `NotificationTargetCertificate::addSpecificTargets()`. Problem is that each certificate may be assigned to a different technician, so each technician may be concerned only by a few items of those mentioned in the notification, and it can be confusing.

3) Replace grouped notifications by unitary notifications. Migrating notification content may be tricky, so we would keep existing grouped notifications during update, but prevent creation of new ones by "hidding" them in the creation form. People would be able to create new unitary notifications if they want to use the technician/group in charge recipients.

I prefer the 3rd solution, but it may be a bit late to fix this in GLPI 10.0.

**Comment by Badkempachi** [comment]
> #6799 was finally a bad idea, as it never worked.
> 
> There are many solutions to this.
> 
>     1. Remove these targets (this is easy to do, but not having these targets may be considered as a missing feature).
> 
>     2. Add all concerned technicians to the notification. It would require to change recipient types from `Notification::ITEM_TECH_IN_CHARGE` to a specific value, so we could handle this in `NotificationTargetCertificate::addSpecificTargets()`. Problem is that each certificate may be assigned to a different technician, so each technician may be concerned only by a few items of those mentioned in the notification, and it can be confusing.
> 
>     3. Replace grouped notifications by unitary notifications. Migrating notification content may be tricky, so we would keep existing grouped notifications during update, but prevent creation of new ones by "hidding" them in the creation form. People would be able to create new unitary notifications if they want to use the technician/group in charge recipients.
> 
> 
> I prefer the 3rd solution, but it may be a bit late to fix this in GLPI 10.0.

Hi, @cedric-anne 
Could you be more specific about what I need to change to make it work correctly?
If I understood everything correctly, then you need to make a change in this line
         $this->addTarget(
             Notification::ITEM_TECH_IN_CHARGE,
             __('Technician in charge of the certificate')
         );
I understand correctly? In my case, it is necessary that notifications come to the “Super-Admin” group or to a specific mailbox, the main thing is that notifications come.

**Comment by cedric-anne** [comment]
> Hi, @cedric-anne
> Could you be more specific about what I need to change to make it work correctly?
> If I understood everything correctly, then you need to make a change in this line
> $this->addTarget(
> Notification::ITEM_TECH_IN_CHARGE,
> __('Technician in charge of the certificate')
> );
> I understand correctly? In my case, it is necessary that notifications come to the “Super-Admin” group or to a specific mailbox, the main thing is that notifications come.

"Technician in charge" recipient cannot work currently. You have to use another type of recipient.

**Comment by Badkempachi** [comment]
> > Hi, @cedric-anne
> > Could you be more specific about what I need to change to make it work correctly?
> > If I understood everything correctly, then you need to make a change in this line
> > $this->addTarget(
> > Notification::ITEM_TECH_IN_CHARGE,
> > __('Technician in charge of the certificate')
> > );
> > I understand correctly? In my case, it is necessary that notifications come to the “Super-Admin” group or to a specific mailbox, the main thing is that notifications come.
> 
> "Technician in charge" recipient cannot work currently. You have to use another type of recipient.

@cedric-anne 
Please clarify instead of "Technician in charge" do I need to specify a profile name? And in this case it will look like:
“$this->addTarget(
Notification::ITEM_TECH_IN_CHARGE,
__('Super-Admin in charge of the certificate')
);” would this be correct?

**Comment by cedric-anne** [comment]
Fix in the code depend on the solution we will choose.

If you want to make it work, you have to change your notification configuration to another recipient type. This has to be done in GLPI UI, not in GLPI source code.

**Comment by Badkempachi** [comment]
> Fix in the code depend on the solution we will choose.
> 
> If you want to make it work, you have to change your notification configuration to another recipient type. This has to be done in GLPI UI, not in GLPI source code.

I changed this setting before and it did not change the situation
![2](https://user-images.githubusercontent.com/97535165/154478885-6b5e968e-93e3-42b8-81a0-eeb9c0f10eaf.png)

**Comment by cedric-anne** [comment]
I tested on my side and I was able to receive the notification by choosing the `Super admin` profile as in your screenshot.

**Comment by cedric-anne** [comment]
> Error inserting e-mail into queue

Where did you see this ? Was it in your mailer log, or in the UI ? Please provide the corresponding exact message.

**Comment by Badkempachi** [comment]
> > Error inserting e-mail into queue
> 
> Where did you see this ? Was it in your mailer log, or in the UI ? Please provide the corresponding exact message.

Hi @cedric-anne ,
The error displayed in the logs was originally attached when the topic was created.
I received this error message in the logs and in the browser, because notifications in the browser and by email were enabled.
To test the operation of the notification about an expired certificate, a certificate was created with an expiration date that ended with the past number, after which the “Automatic action - certificate” was enforced and an error was received at the output.

**Comment by cedric-anne** [comment]
Exact message is `Error inserting email to queue`. I did not find it on my previous analysis as you mentionned a slightly different string.

To investigate on this, we need to have the full log trace corresponding to this message. Can you try to reproduce and check if something is added in your GLPI logs (in `files/_log` directory) ?

**Comment by Badkempachi** [comment]
> Exact message is `Error inserting email to queue`. I did not find it on my previous analysis as you mentionned a slightly different string.
> 
> To investigate on this, we need to have the full log trace corresponding to this message. Can you try to reproduce and check if something is added in your GLPI logs (in `files/_log` directory) ?

@cedric-anne 
The situation has not changed, still the error. The nightly build master.9d53c98.tar.gz was updated today


[2022-02-24 11:03:39] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi/src/DBmysql.php line 367
  *** MySQL query error:
  SQL: INSERT INTO `glpi_queuednotifications` (`itemtype`, `items_id`, `notificationtemplates_id`, `entities_id`, `headers`, `sender`, `sendername`, `name`, `body_text`, `body_html`, `recipient`, `recipientname`, `messageid`, `mode`, `create_time`, `send_time`, `sent_try`, `documents`) VALUES ('Certificate', '-1', '25', '0', '{"Auto-Submitted":"auto-generated","X-Auto-Response-Suppress":"OOF, DR, NDR, RN, NRN"}', 'glpi@localhost.ru', '', '[GLPI] Alarms on expired certificates  Р“РѕР»РѕРІРЅР°СЏ РѕСЂРіР°РЅРёР·Р°С†РёСЏ', 'Entity : Р“РѕР»РѕРІРЅР°СЏ РѕСЂРіР°РЅРёР·Р°С†РёСЏ\n\n\n\nSerial number : \n\nExpiration date : 23-02-2022\n\nhttp://glpi.localhost.ru/index.php?redirect=Certificate_11\n \n\n-- \n\nAutomatically generated by GLPI\n\n', '&#60;!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\"\n                        \'http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\'&#62;&#60;html&#62;\n                        &#60;head&#62;\n                         &#60;META http-equiv=\'Content-Type\' content=\'text/html; charset=utf-8\'&#62;\n                         &#60;title&#62;[GLPI] Alarms on expired certificates  Р“РѕР»РѕРІРЅР°СЏ РѕСЂРіР°РЅРёР·Р°С†РёСЏ&#60;/title&#62;\n                         &#60;style type=\'text/css\'&#62;\n                           \n                         &#60;/style&#62;\n                        &#60;/head&#62;\n                        &#60;body&#62;\n&#60;p&#62;\nEntity : Р“РѕР»РѕРІРЅР°СЏ РѕСЂРіР°РЅРёР·Р°С†РёСЏ&#60;br /&#62;\n\n&#60;br /&#62;Name : test3&#60;br /&#62;\nSerial number : &#60;br /&#62;\nExpiration date : 23-02-2022\n&#60;br /&#62; &#60;a href=\"http://glpi.localhost.ru/index.php?redirect=Certificate_11\"&#62; http://glpi.localhost.ru/index.php?redirect=Certificate_11\n&#60;/a&#62;&#60;br /&#62; &#60;/p&#62;&#60;br&#62;&#60;br&#62;-- \n&#60;br&#62;&#60;br&#62;Automatically generated by GLPI&#60;br&#62;&#60;br&#62;\n\n&#60;/body&#62;&#60;/html&#62;', 'username@localhost.ru', 'Р‘РѕР±СЂРѕРІСЃРєРёР№ РђР»РµРєСЃРµР№', 'GLPI_cELKZRwWpdKNLfEyhbVlvgUcSOhLEsiTlN7YMmLZ-Certificate-N/A.1645689819.1251065330@glpi.localhost.ru', 'mailing', '2022-02-24 11:03:39', '2022-02-24 11:03:39', '0', '')
  Error: Out of range value for column 'items_id' at row 1
  Backtrace :
  src/DBmysql.php:1295                               
  src/CommonDBTM.php:653                             DBmysql->insert()
  src/CommonDBTM.php:1245                            CommonDBTM->addToDB()
  src/NotificationMailing.php:167                    CommonDBTM->add()
  src/Notification.php:608                           NotificationMailing->sendNotification()
  src/NotificationEventAbstract.php:148              Notification::send()
  src/NotificationEvent.php:185                      NotificationEventAbstract::raise()
  src/Certificate.php:750                            NotificationEvent::raiseEvent()
  src/CronTask.php:1016                              Certificate::cronCertificate()
  front/crontask.form.php:50                         CronTask::launch()
  {"user":"9@glpi.localhost.ru"} 
[2022-02-24 11:03:39] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi/src/DBmysql.php line 367
  *** MySQL query error:
  SQL: INSERT INTO `glpi_queuednotifications` (`itemtype`, `items_id`, `notificationtemplates_id`, `entities_id`, `headers`, `sender`, `sendername`, `name`, `body_text`, `body_html`, `recipient`, `recipientname`, `messageid`, `mode`, `create_time`, `send_time`, `sent_try`, `documents`) VALUES ('Certificate', '-1', '25', '0', '{"Auto-Submitted":"auto-generated","X-Auto-Response-Suppress":"OOF, DR, NDR, RN, NRN"}', 'glpi@localhost.ru', '', '[GLPI] Alarms on expired certificates  Р“РѕР»РѕРІРЅР°СЏ РѕСЂРіР°РЅРёР·Р°С†РёСЏ', 'Entity : Р“РѕР»РѕРІРЅР°СЏ РѕСЂРіР°РЅРёР·Р°С†РёСЏ\n\n\n\nSerial Number : \n\nExpiration Date : 23-02-2022\n\nhttp://glpi.localhost.ru/index.php?redirect=Certificate_11\n \n\n-- \n\nAutomatically generated by GLPI\n\n', '&#60;!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\"\n                        \'http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\'&#62;&#60;html&#62;\n                        &#60;head&#62;\n                         &#60;META http-equiv=\'Content-Type\' content=\'text/html; charset=utf-8\'&#62;\n                         &#60;title&#62;[GLPI] Alarms on expired certificates  Р“РѕР»РѕРІРЅР°СЏ РѕСЂРіР°РЅРёР·Р°С†РёСЏ&#60;/title&#62;\n                         &#60;style type=\'text/css\'&#62;\n                           \n                         &#60;/style&#62;\n                        &#60;/head&#62;\n                        &#60;body&#62;\n&#60;p&#62;\nEntity : Р“РѕР»РѕРІРЅР°СЏ РѕСЂРіР°РЅРёР·Р°С†РёСЏ&#60;br /&#62;\n\n&#60;br /&#62;Name : test3&#60;br /&#62;\nSerial Number : &#60;br /&#62;\nExpiration Date : 23-02-2022\n&#60;br /&#62; &#60;a href=\"http://glpi.localhost.ru/index.php?redirect=Certificate_11\"&#62; http://glpi.localhost.ru/index.php?redirect=Certificate_11\n&#60;/a&#62;&#60;br /&#62; &#60;/p&#62;&#60;br&#62;&#60;br&#62;-- \n&#60;br&#62;&#60;br&#62;Automatically generated by GLPI&#60;br&#62;&#60;br&#62;\n\n&#60;/body&#62;&#60;/html&#62;', 'chursin@localhost.ru', 'Р§СѓСЂСЃРёРЅ Р•РІРіРµРЅРёР№', 'GLPI_cELKZRwWpdKNLfEyhbVlvgUcSOhLEsiTlN7YMmLZ-Certificate-N/A.1645689819.851091870@glpi.localhost.ru', 'mailing', '2022-02-24 11:03:39', '2022-02-24 11:03:39', '0', '')
  Error: Out of range value for column 'items_id' at row 1
  Backtrace :
  src/DBmysql.php:1295                               
  src/CommonDBTM.php:653                             DBmysql->insert()
  src/CommonDBTM.php:1245                            CommonDBTM->addToDB()
  src/NotificationMailing.php:167                    CommonDBTM->add()
  src/Notification.php:608                           NotificationMailing->sendNotification()
  src/NotificationEventAbstract.php:148              Notification::send()
  src/NotificationEvent.php:185                      NotificationEventAbstract::raise()
  src/Certificate.php:750                            NotificationEvent::raiseEvent()
  src/CronTask.php:1016                              Certificate::cronCertificate()
  front/crontask.form.php:50                         CronTask::launch()
  {"user":"9@glpi.localhost.ru","mem_usage":"0.008\", 3.85Mio)"}

**Comment by cedric-anne** [comment]
OK

Now the source of the issue is more clear. We have to fix this.

**Comment by cedric-anne** [comment]
Fixed by #10937.
